File: Data Structures and Algorithms Card/Arrays and strings/Examples/Example1/Example1.java
public class Example1 {
    public static void main(String[] args) {
	System.out.println(example1("racecar"));
    }

    public static boolean example1(String s) {
	for (int i = 0, j = s.length() - 1; i < j; ++i, --j) {
	    if (s.charAt(i) != s.charAt(j)) {
		return false;
	    }
	}
	return true;
    }
}

File: Data Structures and Algorithms Card/Arrays and strings/Examples/Example1/example1.cpp
#include <iostream>
#include <string>

bool example1(std::string s);

int main(int argc, char* argv[]) {
  std::cout << std::boolalpha << example1("racecar") << '\n';  
  return 0;
}

bool example1(std::string s) {
  for (int i = 0, j = s.length() - 1; i < j; ++i, --j) {
    if (s[i] != s[j]) {
      return false;
    }
  }
  return true;
}
